<a href="https://colab.research.google.com/github/nurfnick/Numerical_Methods/blob/master/WeBWorK6HomeworkHelper.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# WeBWorK Homework Helper 6  Regression



##WeBWorK 6

Is the number of games won by a major league baseball team in a season related to the team batting average? The table below shows the number of games won and the batting average (in thousandths) of 8 teams.


| Team | Games Won | Batting Average |
|------|-----------|-----------------|
| 1    | 64        | 267             |
| 2    | 61        | 261             |
| 3    | 76        | 279             |
| 4    | 71        | 288             |
| 5    | 97        | 276             |
| 6    | 65        | 278             |
| 7    | 113       | 269             |
| 8    | 120       | 275             |


Using games won as the explanatory variable $x$ find the correlation coefficient and the equation of the least squares line.

In [1]:
from scipy import stats

games = [64,61,76,71,97,65,113,120]
BA = [267, 261,279,288,276,278,269,275]

stats.linregress(games, BA)

LinregressResult(slope=0.015323292057372496, intercept=272.84742052471654, rvalue=0.04304300377517962, pvalue=0.9193939947428843, stderr=0.1452015328046192)

So the correlations coefficient is $r = 0.04304300377517962$ and the equation for the line is $\hat y = 0.015323292057372496 x +272.84742052471654$


## WeBWorK 11

The amounts of 6 restaurant bills and the corresponding amounts of the tips are given in the below.
\begin{array}{c|cccccc}
\mbox{Bill} & 70.29 & 106.27 & 32.98 & 64.30 & 52.44 & 49.72 \cr
\hline
\mbox{Tip}  & 10.00 & 16.00 & 4.50 & 7.70 & 7.00 & 5.28 \cr
\end{array}
Use a 0.05 confidence level to find the following:
The test statistic r = 

Is there a significant correlation?

 1. Yes
 1. No

If the amount of the bill is $40, the best prediction for the amount of the tip is 


In [2]:
bill = [70.29,106.27,32.98,64.30,52.44,49.72]
tip = [10,16,4.50,7.7,7,5.28]

stats.linregress(bill, tip)

LinregressResult(slope=0.16463581370171665, intercept=-1.9038443253075759, rvalue=0.9812002855581488, pvalue=0.0005268217100319339, stderr=0.01619110428907605)

Yes significant correlation, the $p$ value is low!

As for making the prediction.

In [3]:
def f(x):
  return x*0.16463581370171665-1.9038443253075759

In [4]:
f(40)

4.68158822276109